### PR TITLE
Dynamically choose class delimiter for debug entries

### PR DESCRIPTION
**Pull request by zakkak**

### CODE REVIEW

File: substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/Range.java [hunk 1]
```diff
@@ -37,7 +37,7 @@
 
 public class Range {
 
-    private static final String CLASS_DELIMITER = ".";
+    private static final String CLASS_DELIMITER = System.getProperty("os.name").equals("Windows") ? "." : "::";
```
Comment (Karm) [inline substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/Range.java:40]: @zakkak This won't work across different windows versions.    
Use something like 
```
 public static final boolean IS_WINDOWS = System.getProperty("os.name").matches(".*[Ww]indows.*");
```  
That gets rid of variations on "windows server" or "Windows 10" or "Windows" etc...

**Review by Karm — CHANGES_REQUESTED** [review]

File: substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/Range.java (hunk 1 above)
Comment (zakkak) [reply substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/Range.java:40]: `contains` seems enough https://github.com/oracle/graal/blob/d2ddbbb7c079df35892bd78053559047d98cc3cb/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/OS.java#L70

Comment (Karm) [reply substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/Range.java]: @zakkak Correct. I tried on Win 10 and Win 2019 now and `contains` is enough. My regexp was cargoculted from our Noe Core library that goes back to Windows 2003 and JDK 5 so who knows what was it back in the day...and it was written in groovy...

**Review by Karm — APPROVED** [review]
thx

**Pull request merged**